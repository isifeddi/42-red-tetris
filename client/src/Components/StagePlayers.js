import React from "react";

const StagePlayers = ({ stage, user }) => {
  if (!stage) {
    return <span>Loading...</span>;
  }
  return (
    <div style={{ margin: "20px", minWidth: "200px", textAlign: 'center' }}>
      <h3>{user}</h3>
      <table style={{ border: "1px solid" }}>
        {stage.map((row, i) => {
          return (
            <tbody key={`s-${i}`}>
              <tr style={{ height: "15px", padding: "0px", margin: "0px" }}>
                {row.map((cell, x) =>
                  cell[0] === 0 ? (
                    <td
                      style={{
                        border: 0,
                        width: "15px",
                        padding: "0px",
                        margin: "0px",
                        height: "2px",
                        backgroundColor: "#7a0099",
                        opacity: "20%",
                      }}
                      key={`col-${x}`}
                      type={cell[0]}
                    ></td>
                  ) : (
                    <td
                      style={{
                        border: 0,
                        width: "27px",
                        padding: "8px",
                        margin: "0px",
                        height: "2px",
                        backgroundColor: "#b3ffff",
                      }}
                      key={x}
                      type={cell[0]}
                    ></td>
                  )
                )}
              </tr>
            </tbody>
          );
        })}
      </table>
    </div>
  );
};

export default StagePlayers;
