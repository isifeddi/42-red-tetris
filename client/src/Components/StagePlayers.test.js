import renderer from "react-test-renderer";
import StagePlayers from "./StagePlayers";

test("App render test ", () => {
  const tree = renderer.create(<StagePlayers />).toJSON();
  expect(tree).toMatchSnapshot();
});
