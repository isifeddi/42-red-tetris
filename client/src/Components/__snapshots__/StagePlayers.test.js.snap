// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render test  1`] = `
<span>
  Loading...
</span>
`;
